dotnet-client-libraries/YamlDotNet/App/UseCases/BasicSerializeAndDeserialize.cs

```diff
@@ -0,0 +1,19 @@
+using App.Models;
+using YamlDotNet.Serialization;
+
+namespace App.UseCases;
+
+public static class BasicSerializeAndDeserialize
+{
+    public static string SerializeProduct(Product product)
+    {
+        var serializer = new SerializerBuilder().Build();
+        return serializer.Serialize(product);
+    }
+
+    public static Product DeserializeProduct(string yaml)
+    {
+        var deserializer = new DeserializerBuilder().Build();
+        return deserializer.Deserialize<Product>(yaml);
+    }
+}
```

Comment:
Let's combine the `BasicSerializeAndDeserialize` and `ObjectGraphSerializeAndDeserialize` classes into one class that uses generics. Something like this, although I’m not sure what a good name for the class is then.

```suggestion
public static class SerializeAndDeserialize
{
    public static string Serialize<T>(T t)
    {
        var serializer = new SerializerBuilder().Build();
        return serializer.Serialize(t);
    }

    public static T Deserialize<T>(string yaml)
    {
        var deserializer = new DeserializerBuilder().Build();
        return deserializer.Deserialize<T>(yaml);
    }
}
```